AnwesenheitsSimulation/module.php

```diff
@@ -63,6 +64,20 @@ public function ApplyChanges()
             }
         }
 
+        //Transfer legacy SimulationData into attribute
+        $simulationDataID = @$this->GetIDForIdent('SimulationData');
+        $simulationDataAttr = $this->ReadAttributeString('SimulationData');
+        if (IPS_VariableExists($simulationDataID)) {
+            if ($simulationDataAttr == '[]' && function_exists('wddx_deserialize')) {
+                $simulationData = json_encode(wddx_deserialize(GetValue($simulationDataID)));
+                $this->WriteAttributeString('SimulationData', $simulationData);
+                $this->UnregisterVariable('SimulationData');
+            } elseif ($simulationDataAttr == '[]' && !function_exists('wddx_deserialize')) {
+                $this->UpdateData();
```

Comment:
Auch in diesem Fall sollte die Variable gelöscht werden

Reply:
Du löschst aktuell in jedem der drei Fälle die Variable, das kann man sicherlich schöner zusammenführen ;)